src/Mvc/Mvc.ViewFeatures/src/ViewComponentResultExecutor.cs

```diff
@@ -127,9 +137,26 @@ public virtual async Task ExecuteAsync(ActionContext context, ViewComponentResul
                 // IViewComponentHelper is stateful, we want to make sure to retrieve it every time we need it.
                 var viewComponentHelper = context.HttpContext.RequestServices.GetRequiredService<IViewComponentHelper>();
                 (viewComponentHelper as IViewContextAware)?.Contextualize(viewContext);
-
                 var viewComponentResult = await GetViewComponentResult(viewComponentHelper, _logger, result);
-                viewComponentResult.WriteTo(writer, _htmlEncoder);
+
+                if (viewComponentResult is ViewBuffer viewBuffer)
+                {
+                    // In the ordinary case, DefaultViewComponentHelper will return an instance of ViewBuffer. We can simply
+                    // invoke WriteToAsync on it.
+                    await viewBuffer.WriteToAsync(writer, _htmlEncoder);
+                    await writer.FlushAsync();
+                }
+                else
+                {
+                    using var memoryStream = new MemoryStream();
```

Comment:
I'll change this to use the `FileBufferingWriteStream` once that's in.

Reply:
Worth a comment with the issue # to rememberize it?

Reply:
https://github.com/aspnet/AspNetCore/issues/9443